File: YYReader/Parsing/NovelImportResult.swift
import Foundation

struct NovelImportResult: Sendable {
    let bookTitle: String
    let author: String
    /// A stable book-level URL used to deduplicate imports, even when a catalog is absent.
    let sourceBookURL: URL
    let catalogURL: URL
    let hasCatalog: Bool
    let catalog: [ChapterSeed]
    let catalogIsComplete: Bool
    let chapterTitle: String
    let chapterURL: URL
    let bodyText: String
    let previousChapterURL: URL?
    let nextChapterURL: URL?
}

File: windows/YYReader.Windows.Core/Parsing/NovelImportResult.cs
namespace YYReader.Windows.Core.Parsing;

public sealed record NovelImportResult(
    string BookTitle,
    string Author,
    Uri SourceBookUrl,
    Uri CatalogUrl,
    bool HasCatalog,
    IReadOnlyList<ChapterSeed> Catalog,
    bool CatalogIsComplete,
    string ChapterTitle,
    Uri ChapterUrl,
    string BodyText,
    Uri? PreviousChapterUrl,
    Uri? NextChapterUrl);
